### DISCUSSION TITLE
Publish Azure Data Factory - Privacy Questions

### DISCUSSION THREAD

**Discussion by subhraz** [Azure-Player/azure.datafactory.tools#380] [Q&A] [answered]
Hi @NowinskiK,

Posting the questions here as you have suggested. 

I am working on a POC to deploy the ADF pipelines through the Azure DevOps pipeline. Initially I tried the Microsoft approach of doing it using the ARM template from adf_publish branch. Then I came to know about the tool ‘Deploy Azure Data Factory (#adftools)’ developed by you.
I found this tool to be very useful and flexible and thanks a ton to you for coming up with such an intuitive tool.

![image](https://github.com/Azure-Player/azure.datafactory.tools/assets/157047432/2f884e92-a1e2-4aef-97c9-9e4ab65851a6)

Now I would like to use the tool for actual development of my company’s project.
I have a few questions and would appreciate if you could clarify.

1)	What kind of data is collected or stored by the tool or passed outside the organization.
2)	Do the parameters and configuration data placed in the config file e.g., database server name, credentials go out of the organization or stored somewhere else?
3)	Does the code of the ADF instance go out in any form and stored?
4)	Is there a paid version of the tool?

My company needs to follow some stringent compliance requirements and hesitant using the tool before knowing the above.

Thanks again.

**Comment by NowinskiK** [comment, marked as answer]
Hi @subhraz, 
Thanks for using the tool and your kind words about it.
Answering your questions I see you're very conscious about privacy and I'm glad that you're asking about this!

So, let me answer one by one:
ad.1) **NO data is collected** or stored by the tool itself. Also, the tools do not collect any data for analytics purposes.
ad.2) The data is yours! I don't have access to it - there is **NO process sending** these (and any!) data out of your environment. Hence, none of these parameters are shared. The entire code is open-sourced and you can verify that.
ad.3) The code (similarly to configuration) **always stays in your environment**. It's your code, parameters, configuration, and intellectual property that are taken very seriously by me. Again: The tools DO NOT share/send ANY information (files, code, configuration, running times, parameters) ANYWHERE. There is NO NEED to do that.
ad.4) No, there is **no paid version** of the tool. The tool is created by the community for the community. Is open source, free and will always remain free. You can use it commercially with no fees.
I hope that helps - please mark it as an answer when it does.

**Reply by subhraz** [reply]
Thank you for the clarifications.
